compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUConfigureTensorLayouts.cpp

```diff
@@ -292,21 +334,122 @@ setAttentionMatmulAnchor(IREE::GPU::MMAScheduleAttr schedule,
   }
 
-  if (failed(setContractionAnchor(qkSchedule, rewriter, qkMatmul))) {
+  if (failed(setContractionAnchor(qkSchedule, promotedQKOperands, rewriter,
+                                  qkMatmul))) {
     return failure();
   }
 
-  // Do not promote lhs of pvMatmul if we are reusing the intrinsic output.
-  bool promoteLhs = !reuseIntrinsicOutput;
-  bool promoteRhs = true;
-  if (transposeIntrinsic) {
-    std::swap(promoteLhs, promoteRhs);
+  return setContractionAnchor(pvSchedule, promotedPVOperands, rewriter,
+                              pvMatmul);
+}
+
+static IREE::VectorExt::VectorLayoutInterface
+getLayoutForMap(VectorLayoutInterface layout, AffineMap map) {
+  // Project out unusued dims in layout.
+  SmallVector<bool> projectedDims(layout.getRank(), false);
+  for (int dim : getUnusedDimsBitVector(map).set_bits()) {
+    projectedDims[dim] = true;
   }
+  IREE::VectorExt::VectorLayoutInterface projectedLayout =
+      layout.project(projectedDims);
+
+  // Transpose dims in layout.
+  AffineMap permMap = compressUnusedDims(map);
+  SmallVector<int64_t> identity =
+      llvm::to_vector(llvm::seq<int64_t>(permMap.getNumDims()));
+  SmallVector<int64_t> perm = applyPermutationMap<int64_t>(permMap, identity);
+  return projectedLayout.permute(perm);
+}
+
+static LogicalResult setDerivedThreadConfigLayout(
+    IREE::GPU::DerivedThreadConfigAttr config, linalg::LinalgOp linalgOp,
+    ArrayRef<int64_t> workgroupSize, RewriterBase &rewriter) {
 
-  return setContractionAnchor(pvSchedule, rewriter, pvMatmul, promoteLhs,
-                              promoteRhs);
+  int64_t opRank = linalgOp.getNumLoops();
+
+  SmallVector<int64_t> elementTile = config.getStaticTilingLevelSizes(
+      static_cast<unsigned>(IREE::GPU::TilingLevel::Thread), linalgOp);
+
+  SmallVector<int64_t> opShape = linalgOp.getStaticLoopRanges();
+  for (auto [index, size, element] : llvm::enumerate(opShape, elementTile)) {
+    if (ShapedType::isDynamic(size)) {
+      linalgOp->emitError() << "Cannot set layouts for dynamic loop ranges";
+      return failure();
+    }
+
+    if (size % element != 0) {
+      linalgOp->emitError()
+          << "Operation with unsupported number of elements. "
+             "Chosen vector tile sizes for operation are not "
+             "divisible by operation loop ranges at dim: "
+          << index << ", size=" << size << ", vector size = " << element;
+      return failure();
+    }
+
+    size /= element;
+  }
+
+  SmallVector<int64_t> threadTile(opRank, 1);
+  SmallVector<int64_t> threadStrides(opRank, 0);
+
+  int64_t residualThreads = ShapedType::getNumElements(workgroupSize);
+  int64_t currStride = 1;
+
+  for (auto [tile, stride, size] :
+       llvm::reverse(llvm::zip(threadTile, threadStrides, opShape))) {
+    int64_t threadTile;
+    if (residualThreads % size == 0) {
+      threadTile = size;
+    } else if (size % residualThreads == 0) {
+      threadTile = residualThreads;
+    } else {
```

Comment:
Looks/feels kind of similar to the code we have in ConfigureVectorLayout? are we planning to remove the code there?

Reply:
Yes, i updated the patch to remove it.